### PR TITLE
SITL: encourage bash to create a subshell when spawning ardupilot

### PR DESCRIPTION
**Pull request by peterbarker**
This issue is evident in Vagrant if you don't forward X.

_fdm_input_step determined there was no parent and killed ArduPilot.

### CODE REVIEW

**Comment by hamishwillee** [comment]
@rmackay9 Would be good if this could go back into copter 3.3 ....

**Comment by hamishwillee** [comment]
@peterbarker I tried this change out on Cygwin/Windows build and got error (no connection could be made because the target machine actively refused it ..."

This was with ``sim_vehicle.sh -M``

**Comment by lucasdemarchi** [comment]
@peterbarker, is this a fix to #3264?

**Comment by tridge** [comment]
@peterbarker do you have a windows setup to make sure this works on windows?

**Comment by hamishwillee** [comment]
@tridge See https://github.com/diydrones/ardupilot/pull/3325#issuecomment-163868989 - this works on Vagrant but not native Windows builds (at least for the gimbal flag)

**Comment by peterbarker** [comment]
@tridge I do not, sadly.

**Comment by peterbarker** [comment]
@lucasdemarchi It is a fix for the patch referenced in that ticket.  I'll comment on that patch.

**Comment by tridge** [comment]
I just tried this patch on top of master on windows with cygwin and it worked fine.
@hamishwillee can you give a bit more details on what happens when you try it?

**Comment by hamishwillee** [comment]
@tridge So I am running a test to try and simulate gimbal. I'm running on Windows in a native build or a Vagrant build, starting SITL with: 
`sim_vehicle.sh -M`

Without this patch, if I try the above command on Vagrant or on a Cygwin/Windows build the MAVProxy prompt goes though startup but can't connect to SITL - the error displayed is `no connection could be made because the target machine actively refused it ...`

With this patch the 
- Vagrant build does connect as normal. SITL works and I can connect and get messages from the Gimbal
- native build gives me the same error as before (connection refused)
  - If I omit the -M flag connection works (but of course I can't simulate the gimbal.

In summary Windows native build still doesn't work with gimbal simulation enabled. Is that what worked for you?

**Comment by tridge** [comment]
it turns out the gimbal issue is quite separate, and relates to cygwin socket behaviour with blocking sockets
So I will push this change in now, and try to get my head around the cygwin gimbal issue separately

**Comment by lucasdemarchi** [comment]
> @lucasdemarchi It is a fix for the patch referenced in that ticket. I've commented on that patch.

@peterbarker I meant #3264 (not #3074): it seems to be a duplicate of this one.
